src/components/input/input.js

```diff
@@ -381,8 +387,22 @@ function inputTextareaDirective($mdUtil, $window, $mdAria) {
             node.style.minHeight = null;
           }
 
-          var rows = Math.min(min_rows, Math.round(node.scrollHeight / lineHeight));
           node.setAttribute("rows", rows);
+        }
+
+        if (isNaN(rows)) {
+          if (!isNaN(minRows) && contentHeight) {
+            var calcHeight;
+            var minHeight = lineHeight * minRows;
+
+            if (contentHeight <= minHeight) calcHeight = minHeight;
+            else calcHeight = minHeight + (contentHeight - minHeight);
+
+            node.style.height = calcHeight + "px";
+          } else if (contentHeight) {
+            node.style.height = contentHeight + 'px';
```

Comment:
i'd expect a variable to save the height into and applying `node.style.height =` only once for more readability